src/server/pps/server/api_server.go

```diff
@@ -2603,6 +2603,26 @@ func (a *apiServer) RunPipeline(ctx context.Context, request *pps.RunPipelineReq
 	key := path.Join
 	requestProv := make(map[string]*pfs.CommitProvenance)
 	for _, prov := range request.Provenance {
+		if prov == nil {
+			return nil, fmt.Errorf("request should not contain nil provenance")
+		}
+		branch := prov.Branch
+		if branch == nil {
+			if prov.Commit == nil {
+				return nil, fmt.Errorf("request provenance cannot have both a nil commit and nil branch")
+			}
+			provCommit, err := pfsClient.InspectCommit(ctx, &pfs.InspectCommitRequest{
+				Commit: prov.Commit,
+			})
+			if err != nil {
+				return nil, err
+			}
+			branch = provCommit.Branch
+		}
+
+		if prov.Branch.Repo == nil {
+			return nil, fmt.Errorf("request provenance branch must have a non nil repo")
+		}
```

Comment:
Is anything in this code dependent on the code above it? If not I generally like to have validation code at the top of the function before other code so it can exit as fast as possible.